subsys/net/ip/tcp.c

```diff
@@ -1810,18 +1810,31 @@ static bool tcp_validate_seq(struct tcp *conn, struct tcphdr *hdr)
 		(net_tcp_seq_cmp(th_seq(hdr), conn->ack + conn->recv_win) < 0);
 }
 
-static void print_seq_list(struct net_buf *buf)
+static bool check_seq_list(struct net_buf *buf)
 {
+	struct net_buf *last = NULL;
 	struct net_buf *tmp = buf;
 	uint32_t seq;
+	uint32_t next_seq = 0;
+	bool result = true;
 
 	while (tmp) {
 		seq = tcp_get_seq(tmp);
 
 		NET_DBG("buf %p seq %u len %d", tmp, seq, tmp->len);
 
+		if (last != NULL) {
+			if (next_seq != seq) {
+				NET_ERR("Incorrect order in out of order sequence");
```

Comment:
Perhaps print this one in the calling function so that we could print the connection pointer too (might help when debugging issues)

Reply:
Good suggestion, applied